### PR TITLE
Introduce ADR for naming convention of images

### PR DESCRIPTION
**Pull request by pacospace**
Signed-off-by: Francesco Murdaca <fmurdaca@redhat.com>

### CODE REVIEW

File: docs/adr/0004-naming-convention-images.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Project Thoth Release Policy
```
Comment (goern) [inline docs/adr/0004-naming-convention-images.md:1]: This is a naming schema, right?

**Comment by goern** [comment]
I +1 for `curated-stack` as it shows what our intention is: we want to provide a curated/predictable software stack, it might be used by ODH or RHODS or others, it might use S2I or other technology...

**Comment by pacospace** [comment]
> I +1 for `curated-stack` as it shows what our intention is: we want to provide a curated/predictable software stack, it might be used by ODH or RHODS or others, it might use S2I or other technology...

Added to the options and remove final decision!

File: docs/adr/0004-naming-convention-images.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+## Context and Problem Statement
+
+Image names are important for branding and let others identify easily a specific image they need. For example "I want to work on computer vision project with Tensorflow, what stack and image should I use?" Having a trusted well maintained source of images with clean naming convention can help on that.
+
+## Considered Options
+
+* `s2i-{application}` standard name currently, but not everyone knows what S2I is.
+* `odh-{application}` for branding/marketing having ODH in front seems to be the best solution, in this way the name will be shorter as well.
+* `curated-stack-{application}` as it shows what our intention is: we want to provide a curated/predictable software stack, it might be used by ODH or RHODS or others, it might use S2I or other technology.
```
Comment (goern) [inline docs/adr/0004-naming-convention-images.md:16]: during our 'tech talk' discussion there was an argument that the length of the repo name on quay should be considered, so let's go with `ps` and the `predictable stack`

Comment (pacospace) [reply docs/adr/0004-naming-convention-images.md:16]: Perfect! If all good we can merge this PR then and start creating the new repos!

File: docs/adr/0004-naming-convention-images.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
```
Comment (goern) [inline docs/adr/0004-naming-convention-images.md:16]: -> `ps-{.....`

File: docs/adr/0004-naming-convention-images.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+* `ps-{application}` (`ps` stands for `predictable-stack`): showing our intention, being agnostic to technology using the images and building them. Moreover helps from pipeline creation point of view, because the length of repo name on quay can crate issues.
+
+These names are for the core repository name, then `overlays` will allow for combinations of libraries based on other criteria, for example `ml_framework` and/or `hardware`:
+
+    `odh-nlp`
```
Comment (goern) [inline docs/adr/0004-naming-convention-images.md:21]: turn this into `ps-` based example

**Pull request merged**